backend/src/server.ts

```diff
@@ -0,0 +1,26 @@
+import express, { Express, Request, Response } from 'express';
+import bodyParser from 'body-parser';
+import 'dotenv/config';
+import Logger from '@utils/logger';
+import authRouter from '@routes/auth.route';
+import mongoose from 'mongoose';
+
+const app: Express = express();
+app.use(bodyParser.json());
+app.use('/auth', authRouter);
+
+app.get('/', (req: Request, res: Response) => {
+  res.send('Hello, Express!');
+});
+
+mongoose
+  .connect(process.env.MONGODB_URI as string)
+  .then(() => {
+    Logger.info('Connected to MongoDB');
+    app.listen(process.env.PORT, () => {
+      Logger.info('Server is running');
+    });
+  })
+  .catch((error) => {
+    Logger.error(`Error connecting to MongoDB: ${error}`);
+  });
```

Comment:
Configure this within a dbConfig.ts file:
```suggestion
connectToDatabase()
  .then(() => {
    app.listen(process.env.PORT, () => {
      Logger.info('Server is running');
    });
  })
  .catch((error) => {
    Logger.error(`Error starting server: ${error}`);
    process.exit(1);
  });
```
```

// dbConfig.ts
import mongoose from 'mongoose';
import Logger from '@utils/logger';

const connectToDatabase = async (): Promise<void> => {
  try {
    await mongoose.connect(process.env.MONGODB_URI as string);
    Logger.info('Connected to MongoDB');
  } catch (error) {
    Logger.error(`Error connecting to MongoDB: ${error}`);
    throw error; 
  }
};

export default connectToDatabase;

```